src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -643,9 +643,12 @@ static inline int _xccdf_policy_rule_generate_fix_ansible(const char *template,
 			}
 		}
 		else {
-			char *remediation_part = malloc((ovector[0] + 1) * sizeof(char));
-			memcpy(remediation_part, &fix_text[start_offset], ovector[0]);
-			remediation_part[ovector[0]] = '\0';
+			// Remarks: ovector doesn't contain values relative to start_offset, it contains
+			// absolute indices of fix_text.
+			const int length_between_matches = ovector[0] - start_offset;
+			char *remediation_part = malloc((length_between_matches + 1) * sizeof(char));
+			memcpy(remediation_part, &fix_text[start_offset], length_between_matches);
+			remediation_part[length_between_matches] = '\0';
```

Comment:
I noticed some lines with just indentation is left where the ` #promote to variable` task was.
As it doesn't break syntax, and it looks like it is common in other places of the generated playbook, I consider it a non blocker.